Sources/Networking/SessionDelegate.swift

```diff
@@ -145,6 +146,15 @@ class SessionDelegate: NSObject {
         lock.unlock()
         task?.forceCancel()
     }
+    
+    /// You could set the extra handler before a downloading task starts.
+    func setExtraHandler(_ handler: URLSessionDataDelegate?) {
+        extraHandler = handler
+    }
+    func getExtraHandler() -> URLSessionDataDelegate? {
+        return extraHandler
+    }
+    
```

Comment:
I prefer to remove these methods and make `extraHandler` to `internal`, so we can just use the property without this boilerplate method calling.